Claudia/src/main/java/com/endava/task/second/User.java

```diff
@@ -0,0 +1,46 @@
+package com.endava.task.second;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class User {
+    private static long countId = 1;
+    public long id = 1;
+    public String username;
+    public Map<String, Address> addressMap;
+
+    public User(String username, String addressType, Address address) {
+        this.id = countId++;
+        this.username = username;
+        this.addressMap = addAddress(addressType, address);
+
+    }
+
+    public Map<String, Address> addAddress(String type, Address address) {
+        Map<String, Address> newAddress = new HashMap<>();
+        newAddress.put(type, address);
+        return newAddress;
+    }
+
+    public void printUser() {
+        System.out.println("User details are : ");
+        System.out.println("Id = " + this.id);
+        System.out.println("Username = " + this.username);
+        System.out.println("Addresses are:");
+        for (Map.Entry<String, Address> addressEntry : this.addressMap.entrySet()) {
+            System.out.println(addressEntry.getKey());
+            addressEntry.getValue().printAddress();
+        }
+    }
+
+    public boolean containsChisinau() {
+        boolean containsChisinau = false;
+        for (Map.Entry<String, Address> addressEntry : this.addressMap.entrySet()) {
```

Comment:
Imagine if the first set of values in this map will contain Chisinau but the second one will not -> this method will return false even if there is address where exists city like Chisinau.

Reply:
Done

Reply:
Try to use the same approach as in your containsClothes() method. If condition (addressEntry.getValue().containsChisinau()) matches -> return true (so you will be sure that there is at least one address with Chisianau city).

Reply:
You're right, changed.